cmd/exo/cmd/request.go

```diff
@@ -2,10 +2,114 @@ package cmd
 
 import (
 	"fmt"
+	"math/rand"
+	"sync"
+	"time"
 
 	"github.com/exoscale/egoscale"
+	"github.com/vbauerster/mpb"
+	"github.com/vbauerster/mpb/decor"
 )
 
+type task struct {
+	egoscale.AsyncCommand
+	string
+}
+
+// asyncTasks message variable must have same size with cmds
+func asyncTasks(tasks []task) ([]interface{}, []error) {
+
+	//init results
+	errors := &[]error{}
+	r := make([]interface{}, len(tasks))
+	responses := &r
+
+	//create task Progress
+	taskBars := make([]*mpb.Bar, len(tasks))
+	maximum := 10
+	var wg sync.WaitGroup
+	p := mpb.New(mpb.WithWaitGroup(&wg), mpb.WithContext(gContext), mpb.WithWidth(40))
+	wg.Add(len(tasks))
+
+	//exec task and init bars
+	for i, task := range tasks {
+		c := make(chan []int)
+		go execTask(task, i, c, responses, errors)
+		taskBars[i] = p.AddBar(int64(maximum),
+			mpb.PrependDecorators(
+				// simple name decorator
+				decor.Name(task.string),
+				// decor.DSyncWidth bit enables column width synchronization
+				decor.Percentage(decor.WCSyncSpace),
+			),
+			mpb.AppendDecorators(
+				// replace ETA decorator with "done" message, OnComplete event
+				decor.OnComplete(
+					// ETA decorator with ewma age of 60
+					decor.EwmaETA(decor.ET_STYLE_GO, 60), "done",
+				),
+			),
+		)
+
+		//listen for bar progress
+		go func(chanel chan []int) {
+			defer wg.Done()
+			defer close(chanel)
+			max := 100 * time.Millisecond
+			count := 1
+			for status := range chanel {
```

Comment:
Coco?

Reply:
coco chanel ?

Reply:
![](https://memegenerator.net/img/instances/61374337/im-in-love-with-the-coco-chanel-no-5.jpg)

Reply:
ahah